library/core/src/num/uint_macros.rs

```diff
@@ -1979,6 +1979,41 @@ macro_rules! uint_impl {
             acc * base
         }
 
+        /// Returns the square root of the number, rounded down.
+        ///
+        /// # Examples
+        ///
+        /// Basic usage:
+        /// ```
+        #[doc = concat!("assert_eq!(10", stringify!($SelfT), ".isqrt(), 3);")]
+        /// ```
+        #[stable(feature = "isqrt", since = "1.73.0")]
+        #[rustc_const_stable(feature = "isqrt", since = "1.73.0")]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline]
+        pub const fn isqrt(self) -> Self {
```

Comment:
Python's `isqrt` comes with this insightful note:

[_"For some applications, it may be more convenient to have the least integer **a** such that **n ≤ a²**, or in other words the ceiling of the exact square root of **n**. For positive **n**, this can be computed using `a = 1 + isqrt(n - 1)`."_](https://docs.python.org/3/library/math.html#math.isqrt)

Do you think it would be beneficial to incorporate something like this, or is it easy enough to figure out this solution for the people who might need it?

Reply:
Personally I think this is a "standard trick" and anybody needing to round up is probably aware of it. Also, a similar remark would apply to `ilog`, `ilog2` and `ilog10` as well.